lib/src/obsoleted/analyzer_plugin/analyzer_plugin.dart

```diff
@@ -368,40 +368,41 @@ class MetricsAnalyzerPlugin extends ServerPlugin {
     ScopedFunctionDeclaration function,
     InternalResolvedUnitResult source,
     AnalyzerPluginConfig config,
-  ) {
-    final controlFlowAstVisitor = CyclomaticComplexityFlowVisitor();
-
-    function.declaration.visitChildren(controlFlowAstVisitor);
-
-    final cyclomaticLines = controlFlowAstVisitor.complexityEntities
-        .map((entity) => nodeLocation(node: entity, source: source).start.line)
-        .toSet();
-
-    return UtilitySelector.functionReport(
-      FunctionRecord(
-        location: nodeLocation(
-          node: function.declaration,
-          source: source,
-        ),
-        metrics: const [],
-        argumentsCount: getArgumentsCount(function),
-        cyclomaticComplexityLines: Map.fromEntries(cyclomaticLines.map(
-          (lineIndex) => MapEntry(
-            lineIndex,
-            controlFlowAstVisitor.complexityEntities
-                .where((entity) =>
-                    nodeLocation(node: entity, source: source).start.line ==
-                    lineIndex)
-                .length,
+  ) =>
+      UtilitySelector.functionReport(
+        Report(
+          location: nodeLocation(
+            node: function.declaration,
+            source: source,
           ),
-        )),
-        linesWithCode: List.unmodifiable(<int>[]),
-        operators: Map.unmodifiable(<String, int>{}),
-        operands: Map.unmodifiable(<String, int>{}),
-      ),
-      config.metricsConfigs,
-    );
-  }
+          metrics: [
+            CyclomaticComplexityMetric(config: {
+              CyclomaticComplexityMetric.metricId:
```

Comment:
Why do we need to pass a metric Id to its own metric?

Reply:
I will rewrite this code block in next time